### ISSUE TITLE
yarprobotinterface silently suppresses groups in which one parameters list is not ending with a parenthesis

### ISSUE THREAD

**Issue by traversaro** [robotology/yarp#832]
Example: 

```
  <device name="wholebodydynamics" type="wholebodydynamics">
        <param name="modelFile">model.urdf</param>
        <param name="fixedFrameGravity">(0,0,-9.81)</param>

        <group name="FOO">
            <param name="bar">(1.0,2.0,
                               3.0,4.0</param>
        </group>
  </device>
```

In this case the `FOO` group is simply not passed to the `open` of the device, without any printed warning or error. If the parser encounter an error of this kind, it should fail as soon as possible to simplify debugging of malformed configuration files. 

@DanielePucci @fjandrad

**Comment by DanielePucci** [comment]
-.-
~~~
without any printed warning or error
~~~

**Comment by pattacini** [comment]
I'm having big problems launching the `cartesiancontrollerserver` components along with the simulator.

If I type:
```
$ yarprobotinterface --context simCartesianControl --config no_legs.xml
```

I get:
```
[ERROR]SolverNameToConnect option is missing
```

The option is there.

@drdanz has this error something to do with this patch?

**Comment by Nicogene** [comment]
Which is the status of this issue? Has been fixed by #995?

**Comment by traversaro** [comment]
I think so, as far as I remember.
